### PR TITLE
TRT-2692: Agentic Jira solver eval cases

### PR DESCRIPTION
**Pull request by smg247**
Adds the eval case definitions for the jira-solver benchmark ([TRT-2692](https://redhat.atlassian.net/browse/TRT-2692)). Five historical JIRA issues with known-good solutions, covering a range of difficulty and languages:

  | Case | JIRA | Difficulty | What it tests |
  |------|------|-----------|---------------|
  | 001 | [TRT-2698](https://redhat.atlassian.net/browse/TRT-2698) | Medium | Makefile + Python + Dockerfile (mcp tests) |
  | 002 | [TRT-2753](https://redhat.atlassian.net/browse/TRT-2753) | Easy | Single Go file (routing fix) |
  | 003 | [TRT-2660](https://redhat.atlassian.net/browse/TRT-2660) | Easy-Medium | Go + JS (null serialization crash) |
  | 004 | [TRT-2678](https://redhat.atlassian.net/browse/TRT-2678) | Medium | Python (new helper + tests) |
  | 005 | [TRT-2795](https://redhat.atlassian.net/browse/TRT-2795) | Easy | Single JS file (filter comparison bug) |

  Each case includes:
  - `input.yaml` — repo, base/expected branch names for `openshift-trt/sippy-eval`
  - `annotations.yaml` — difficulty, expected files changed, key signals
  - `jira-issue.json` — snapshot of the JIRA REST API response

  The companion PR in openshift/release adds the eval workflow (eval-init, eval-judge steps) and CI operator config. The eval repo (`openshift-trt/sippy-eval`) needs to be created separately with
  base/expected branches pushed from sippy's git history.

## Summary by CodeRabbit

* **New Features**
  * Added five Jira Solver evaluation cases covering MCP-related tests, `/sippy-ng/` trailing-slash routing, null explanation handling, backend readiness verification, and variant filter behavior.
  * Included per-case configuration, Jira issue fixtures, and expected-change expectations to support end-to-end evaluation runs.
* **Documentation**
  * Expanded the Jira solver eval README with the full workflow, case structure (base vs expected), how to add new cases, how to run/judge, and where judge artifacts are produced.

### CODE REVIEW

File: evals/jira-solver/README.md [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+   ```
+6. Create `input.yaml` and `annotations.yaml` following the existing cases
+
+## Running the eval
+
+Trigger via Gangway with case and model overrides:
+
+```bash
+# Run a specific case with a specific model
+gangway trigger --job periodic-ci-openshift-release-main-jira-solver-eval-jira-solver-eval \
```
[the remark this answers is not in the document]
Comment (smg247) [reply evals/jira-solver/README.md:67]: The generated job name actually is periodic-ci-openshift-release-main-jira-solver-eval-jira-solver-eval — the double suffix is correct

File: evals/jira-solver/cases/case-002-trt-2753-trailing-slash/jira-issue.json [hunk 2]
```diff
@@ -0,0 +1 @@
...o trailing slash) fails to render the React app. The issue only manifests in certain situations such as first page load, suggesting a race between server-side path rewriting and client-side React Router initialization.\n\n\n\nh2. Acceptance Criteria\n\nAll routes continue to work, and the non-trailing slash version works to for the FIRST page visit.  Please confirm with playwright.\n\n---\nRequested via internal bot","comment":{"comments":[],"self":"https://redhat.atlassian.net/rest/api/2/issue/6284195/comment","maxResults":0,"total":0,"startAt":0},"priority":{"self":"https://redhat.atlassian.net/rest/api/2/priority/10003","iconUrl":"https://redhat.atlassian.net/images/icons/priorities/medium.svg","name":"Normal","id":"10003"},"status":{"self":"https://redhat.atlassian.net/rest/api/2/status/6","description":"The work item is considered finished, the resolution is correct. work items which are closed can be reopened.","iconUrl":"https://redhat.atlassian.net/images/icons/statuses/closed.png","name":"Closed","id":"6","statusCategory":{"self":"https://redhat.atlassian.net/rest/api/2/statuscategory/3","id":3,"key":"done","colorName":"green","name":"Done"}},"labels":["issue-for-agent"]}}
```
[the remark this answers is not in the document]
Comment (smg247) [reply evals/jira-solver/cases/case-002-trt-2753-trailing-slash/jira-issue.json:1]: The JSON file has one object. @coderabbitai, can you double-check this?
